.github/workflows/upload_component.yml

```diff
@@ -10,18 +10,56 @@ on:
 jobs:
   upload_components:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        bsp: [
+          esp32_azure_iot_kit,
+          esp32_s2_kaluga_kit,
+          esp_wrover_kit,
+          esp-box,
+          esp-box-3,
+          esp-box-lite,
+          esp32_s3_usb_otg,
+          esp32_s3_eye,
+          esp32_s3_lcd_ev_board,
+          esp32_s3_korvo_1,
+          esp32_s3_korvo_2,
+          esp32_lyrat,
+          esp32_c3_lcdkit,
+          esp_bsp_generic,
+          esp_bsp_devkit,
+          esp32_p4_function_ev_board,
+          m5stack_core,
+          m5stack_core_2,
+          m5stack_core_s3,
+          m5_atom_s3,
+          m5dial,
+        ]
     steps:
       - uses: actions/checkout@master
 
-      - name: Upload components to component service
+      - name: Prepare BSP examples
+        run: |
+          pip install idf-component-manager==1.* pyyaml
+          python .github/ci/bsp_example_update.py --bsp=${{ matrix.bsp }}
+
+      - name: Upload BSP components
+        uses: espressif/upload-components-ci-action@v1
+        with:
+          directories: >
+            bsp/${{ matrix.bsp }}
+          namespace: "espressif"
+          api_token: ${{ secrets.IDF_COMPONENT_API_TOKEN }}
+          dry_run: ${{ github.ref_name != 'master' || github.repository_owner != 'espressif' }}
+
+      - name: Upload Other components
```

Comment:
I believe the `Upload Other components` step is going to be triggered for each bsp matrix option (e.g. 21x).

A solution would be to extract it into a separate job

Reply:
You are right, thanks. I changed.